substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/ResourceAttributesView.java

```diff
@@ -0,0 +1,117 @@
+package com.oracle.svm.core.jdk;
+
+import java.io.IOException;
+import java.nio.file.ReadOnlyFileSystemException;
+import java.nio.file.attribute.BasicFileAttributeView;
+import java.nio.file.attribute.FileAttributeView;
+import java.nio.file.attribute.FileTime;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class ResourceAttributesView implements BasicFileAttributeView {
+
+    private enum AttributeID
+    {
+        size,
+        creationTime,
+        lastAccessTime,
+        lastModifiedTime,
+        isDirectory,
+        isRegularFile,
+        isSymbolicLink,
+        isOther,
+        fileKey,
+    }
+
+    private final ResourcePath path;
+    private final boolean isBasic;
+
+    public ResourceAttributesView(ResourcePath path, boolean isBasic) {
+        this.path = path;
+        this.isBasic = isBasic;
+    }
+
+    @SuppressWarnings("unchecked")
+    static <V extends FileAttributeView> V get(ResourcePath path, Class<V> type) {
+        if (type == null)
+            throw new NullPointerException();
+        if (type == BasicFileAttributeView.class)
+            return (V) new ResourceAttributesView(path, true);
+        if (type == ResourceAttributesView.class)
+            return (V) new ResourceAttributesView(path, false);
+        return null;
+    }
+
+    static ResourceAttributesView get(ResourcePath path, String type) {
+        if (type == null)
+            throw new NullPointerException();
+        if (type.equals("basic"))
+            return new ResourceAttributesView(path, true);
+        if (type.equals("resource"))
+            return new ResourceAttributesView(path, false);
+        return null;
+    }
+
+    @Override
+    public String name() {
+        return isBasic ? "basic" : "resource";
+    }
+
+    @Override
+    public ResourceAttributes readAttributes() throws IOException {
+        return path.getAttributes();
+    }
+
+    Map<String, Object> readAttributes(String attributes) throws IOException {
+        ResourceAttributes resourceAttributes = readAttributes();
+        LinkedHashMap<String, Object> map = new LinkedHashMap<>();
+        if ("*".equals(attributes)) {
+            for (AttributeID id : AttributeID.values()) {
+                try {
+                    map.put(id.name(), attribute(id, resourceAttributes));
+                } catch (IllegalArgumentException x) {
+                    // Silently handle this exception.
```

Comment:
Hmmmm, this is almost always a bad idea. When does this happen?

Reply:
Can we use something other than exceptions for control flow?

Reply:
We can try with an if-else statement instead of a try-catch. Sounds okay?

Reply:
We are now checking in an if-else statement if the specified attribute is in an attribute set instead of throwing an exception.